test/testdata/infer/generics/initialize_class_bound.rb

```diff
@@ -0,0 +1,62 @@
+# typed: strict
+class Module; include T::Sig; end
+
+module Thing
+  extend T::Helpers
+  interface!
+
+  sig {abstract.returns(Integer)}
+  def foo; end
+
+  module Factory
+    extend T::Generic
+    interface!
+
+    has_attached_class!(:out) { {upper: Thing} }
```

Comment:
We don't seem to have any tests for `has_attached_class!` with bounds, but the enclosing module is used in a context that doesn't satisfy those bounds.  Can we add a testcase like that, please?  (I assume that the error messages will be somewhat sensible, but maybe not!)